src/env/as_env.rs

```diff
@@ -130,3 +130,32 @@ pub fn assembly_script_date_now(mut ctx: FunctionEnvMut<ASEnv>) -> ABIResult<f64
     let ret = utime as f64;
     Ok(ret)
 }
+
+#[named]
+pub fn assembly_script_console_log(
+    mut ctx: FunctionEnvMut<ASEnv>,
+    message: StringPtr,
+) -> ABIResult<()> {
+    let env = ctx.data().clone();
+    if cfg!(not(feature = "gas_calibration")) {
+        sub_remaining_gas_abi(&env, &mut ctx, function_name!())?;
+    }
+
+    let memory = ctx
+        .data()
+        .get_wasm_env()
+        .memory
+        .as_ref()
+        .expect("Failed to get memory on env")
+        .clone();
+    let message_ = message
+        .read(&memory, &ctx)
+        .map_err(|e| wasmer::RuntimeError::new(e.to_string()));
+
+    if message_.is_err() {
+        abi_bail!("aborting failed to load message");
+    }
+
+    println!("[console] {}", message_.unwrap());
```

Comment:
So if I understand correctly this is bound to appear in the node logs? Do we really have a need for that?

Reply:
I think the SC developer prefer to have it on his side and he don't necessary have a node running.

Reply:
We already have a assembly_script_abi_print that have the same behavior and by implementing this we improve our assembly script compatibility.

Reply:
I think `print` should be removed it was used before `generateEvent` now we should only use `generateEvent` that allow developer to fetch the events on his side. I think `console.log` should have this behavior

Reply:
> I think the SC developer prefer to have it on his side and he don't necessary have a node running. It should be better on developer side.

Okay but its pretty much the same here, print is implemented in massa and logs the message with the tracing crate, and this uses println and no interface function so I fail to see the difference here

> We already have a assembly_script_abi_print that have the same behavior and by implementing this we improve our assembly script compatibility.

I do agree with this though

Reply:
Also PR just got closed because it pointed towards testnet_18